packages/async-flow/test/async-flow-debug.test.js

```diff
@@ -0,0 +1,219 @@
+/* eslint-env node */
+// eslint-disable-next-line import/order
+import { test, getBaggage, annihilate } from './prepare-test-env-ava.js';
+
+import { passStyleOf } from '@endo/pass-style';
+import { makePromiseKit } from '@endo/promise-kit';
+import { eventLoopIteration } from '@agoric/internal/src/testing-utils.js';
+import { isVow } from '@agoric/vow/src/vow-utils.js';
+import { prepareVowTools } from '@agoric/vow';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+
+import { prepareTestAsyncFlowTools } from './_utils.js';
+import { consoleRecorder, originalConsole } from './console-recorder-shim.js';
+
+/**
+ * @import {Zone} from '@agoric/base-zone';
+ * @import {ExecutionContext} from 'ava';
+ */
+
+/**
+ * @param {ExecutionContext<unknown>} t
+ * @param {any} hOrch
+ * @param {any} reason
+ * @param {any[]} [recording]
+ */
+const rejectionRecordingAssertions = (t, hOrch, reason, recording) => {
+  assert(recording);
+
+  // console.log(
+  //   '@@@ recording was:',
+  //   recording.map(({ method, args }, i) => ({ i, method, args })),
+  // );
+  const cons = (method, ...args) => ({
+    method,
+    args,
+  });
+
+  t.true(recording.every(entry => entry.thisArg === originalConsole));
+
+  // The recording must have at least the entries we have in the expected
+  // pattern below.
+  const recLen17 = Math.max(recording.length, 17);
+
+  // Check the guestMethod stack trace exists.
+  const indexToStackRegex = {
+    2: /\bdoReject\b/,
+    8: /\sIn "reject" method of \(Orchestra\)\s/,
+    [recLen17 - 5]: /\basync-flow-debug\b/,
+    [recLen17 - 2]: /\basyncFlow\b.*\btestRejectionPlay\b/s,
+  };
+
+  const consStack = i => {
+    const re = indexToStackRegex[i];
+    delete indexToStackRegex[i];
+    t.truthy(re, `recorded entry ${i} has a stack regex`);
```

Comment:
Why not accept the `re` as an argument instead of this definition indirection?

```suggestion
  const consStack = (i, re) => {
    t.truthy(re, `recorded entry ${i} has a stack regex`);
```